pennylane_lightning/lightning_qubit.py

```diff
@@ -576,6 +576,33 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
 
         return M.probs(device_wires)
 
+    def generate_samples(self):
+        """Generate samples
+
+        Returns:
+            Array of samples in binary
+        """
+
+        # To support np.complex64 based on the type of self._state
+        dtype = self._state.dtype
+        if dtype == np.complex64:
+            use_csingle = True
+        elif dtype == np.complex128:
+            use_csingle = False
+        else:
+            raise TypeError(f"Unsupported complex Type: {dtype}")
+
+        # Initialization of state
+        ket = np.ravel(self._state)
+
+        if use_csingle:
+            ket = ket.astype(np.complex64)
+
+        state_vector = StateVectorC64(ket) if use_csingle else StateVectorC128(ket)
+        M = MeasuresC64(state_vector) if use_csingle else MeasuresC128(state_vector)
+
+        return M.generate_samples(len(self.wires), self.shots).astype(int)
```

Comment:
Sorry! I just noticed `astype(int)`. Shouldn't be `complex` to become compatible with the return value of `generate_samples` method in `QubitDevice`? lmk if I'm missing something.

Reply:
that is a typo in the pennylane documentation, it actually returns an int array if you look at states_to_binary which is returned at the bottom of generate_samples